src/main/java/com/aws/greengrass/FleetProvisioningByClaimPlugin.java

```diff
@@ -140,19 +144,38 @@ public ProvisionConfiguration updateIdentityConfiguration(ProvisionContext provi
                     "Caught exception while establishing connection to AWS Iot");
 
             IotIdentityHelper iotIdentityHelper = iotIdentityHelperFactory.getInstance(connection);
-            CreateKeysAndCertificateResponse response;
-            response = FutureExceptionHandler.getFutureAfterCompletion(iotIdentityHelper.createKeysAndCertificate(),
-                    "Caught exception during PublishCreateKeysAndCertificate");
 
-            writeCertificateAndKeyToPath(response, parameterMap.get(ROOT_PATH_PARAMETER_NAME).toString());
+            String certificateOwnershipToken = null;
+            if (csrPath == null) {
+                CreateKeysAndCertificateResponse response;
+                response = FutureExceptionHandler.getFutureAfterCompletion(iotIdentityHelper.createKeysAndCertificate(),
+                        "Caught exception during PublishCreateKeysAndCertificate");
+
+                writeCertificateAndKeyToPath(response, parameterMap.get(ROOT_PATH_PARAMETER_NAME).toString());
+                certificateOwnershipToken = response.certificateOwnershipToken;
+            } else {
+                Path csrFile = Paths.get(csrPath);
+                if (!csrFile.toFile().exists()) {
+                    logger.error("CSR file not found at " + csrPath);
+                    throw new InterruptedException("CSR file not found.");
```

Comment:
Please use a more appropriate exception type. The current implementation before this PR uses a RuntimeException (https://github.com/aws-greengrass/aws-greengrass-fleet-provisioning-by-claim/blob/a057af8177bf4ef46839ea6cfc94e15e4f4ffede/src/main/java/com/aws/greengrass/FleetProvisioningByClaimPlugin.java#L181) when the inputs are improper. Therefore, this should also use a runtime exception which includes a good error message.